tools/Propulsion.Tool/Program.fs

```diff
@@ -24,14 +24,30 @@ type Parameters =
 
 and [<NoComparison; NoEquality>] InitAuxParameters =
     | [<AltCommandLine("-ru"); Mandatory>]  Rus of int
+    | [<AltCommandLine "-A">]               Autoscale
+    | [<AltCommandLine "-m">]               Mode of CosmosModeType
     | [<AltCommandLine("-s")>]              Suffix of string
     | [<CliPrefix(CliPrefix.None)>]         Cosmos of ParseResults<Args.Cosmos.Parameters>
     interface IArgParserTemplate with
         member a.Usage = a |> function
             | Rus _ ->                      "Specify RU/s level to provision for the Aux Container."
+            | Autoscale ->                  "Autoscale provisioned throughput. Use --rus to specify the maximum RU/s."
+            | Mode _ ->                     "Configure RU mode to use Container-level RU, Database-level RU, or Serverless allocations (Default: Use Container-level allocation)."
             | Suffix _ ->                   "Specify Container Name suffix (default: `-aux`)."
             | Cosmos _ ->                   "Cosmos Connection parameters."
 
+and CosmosInitInfo(args : ParseResults<InitAuxParameters>) =
+    member _.ProvisioningMode =
+        let throughput () =
+            if args.Contains Autoscale
+            then Equinox.CosmosStore.Core.Initialization.Throughput.Autoscale (args.GetResult(Rus, 4000))
+            else Equinox.CosmosStore.Core.Initialization.Throughput.Manual (args.GetResult(Rus, 400))
+        match args.GetResult(Mode, CosmosModeType.Container) with
+        | CosmosModeType.Container ->       Equinox.CosmosStore.Core.Initialization.Provisioning.Container (throughput ())
+        | CosmosModeType.Db ->              Equinox.CosmosStore.Core.Initialization.Provisioning.Database (throughput ())
+        | CosmosModeType.Serverless ->
+            if args.Contains Rus || args.Contains Autoscale then raise (failwith "Cannot specify RU/s or Autoscale in Serverless mode")
```

Comment:
raise failwith is redundant (`failwith` === `raise << sprintf`)
but to work correctly, this should actually raise MissingArg by means of https://github.com/jet/propulsion/blob/master/tools/Propulsion.Tool/Args.fs#L8. to pass through the catches in `main` and render neatly (I think that may be broken atm - I dont mind if you manage to debug and fix that or not)

Reply:
sorry, i have fixed this to use MissingArg.

Reply:
i tested the validation after using 'MissingArg' and it worked fine.

Reply:
Thanks.

Yeah, the bug I was talking about was me using e.g. `p.GetValue SomethingMandatory` - if you call that inside the main body of the code, an ArguParseException get emitted, which was being logged with a stacktrace which uglified it a lot

(The `missingArg` vs `failwith` is important for the same reason to bypass that filter)

🤔 A PR to Argu to add a `missingArg` function that `raise`s an `ArguParseException` would probably be reasonable and would make the error handling much cleaner

Reply:
Ahh ok. Could we use `p.TryGetResult` and wire it up a with func like `requireArg` which can internally raise the `MissingArg` exp?

Reply:
Hopefully something clean will arise from https://github.com/fsprojects/Argu/issues/175
In general adding indirection and 'doing things properly' are not always the answer for arg parsing - being able to read the logic is more valuable

e.g. not sure that would help this code: https://github.com/jet/equinox/blob/b7e220750369a2a5d5645ddf9dd14455bf4dc2de/tools/Equinox.Tool/Program.fs#L86-L89
given that the calls to `GetResult` on L87 can trigger an exception (after the initial top level argu parse), in the same code path

...and the fact the code is terse has just helped me spot a `&&` that should be a `||` !

@enricosada turned me on to this point over time - the use of `member val` to force eager triggering of mandatory argument checks is also part of having all this be predictable and terse.

Reply:
Sorry to necropost, but I eventually discovered that `p.Raise` is how you do Raise an ArguParseException, and you don't need a `MissingArg` at all!